src/modules/workspaces/workspaces.usecase.spec.ts

```diff
@@ -949,4 +950,441 @@ describe('WorkspacesUsecases', () => {
+      });
+
+      it('When user is not part of workspace, then it should throw', async () => {
+        const user = newUser();
+        const workspace = newWorkspace();
+        const team = newWorkspaceTeam({
+          workspaceId: workspace.id,
+        });
+
+        jest.spyOn(teamRepository, 'getTeamById').mockResolvedValue(team);
+        jest.spyOn(workspaceRepository, 'findOne').mockResolvedValue(workspace);
+        jest
+          .spyOn(workspaceRepository, 'findWorkspaceUser')
+          .mockResolvedValue(null);
+
+        await expect(
+          service.changeTeamManager(team.id, user.uuid),
+        ).rejects.toThrow(BadRequestException);
+      });
+
+      it('When user is not part of team, then it should throw', async () => {
+        const user = newUser();
+        const workspace = newWorkspace();
+        const workspaceUser = newWorkspaceUser({ workspaceId: workspace.id });
+        const team = newWorkspaceTeam({
+          workspaceId: workspace.id,
+        });
+
+        jest.spyOn(teamRepository, 'getTeamById').mockResolvedValue(team);
+        jest.spyOn(workspaceRepository, 'findOne').mockResolvedValue(workspace);
+        jest
+          .spyOn(workspaceRepository, 'findWorkspaceUser')
+          .mockResolvedValue(workspaceUser);
+        jest.spyOn(teamRepository, 'getTeamUser').mockResolvedValue(null);
+
+        await expect(
+          service.changeTeamManager(team.id, user.uuid),
+        ).rejects.toThrow(BadRequestException);
+      });
+
+      it('When user is being assigned succesfully, then it should resolve', async () => {
+        const user = newUser();
+        const workspace = newWorkspace();
+        const workspaceUser = newWorkspaceUser({ workspaceId: workspace.id });
+        const team = newWorkspaceTeam({
+          workspaceId: workspace.id,
+        });
+        const teamUser = newWorkspaceTeamUser({
+          teamId: team.id,
+          memberId: user.uuid,
+        });
+
+        jest.spyOn(teamRepository, 'getTeamById').mockResolvedValue(team);
+        jest.spyOn(workspaceRepository, 'findOne').mockResolvedValue(workspace);
+        jest
+          .spyOn(workspaceRepository, 'findWorkspaceUser')
+          .mockResolvedValue(workspaceUser);
+        jest.spyOn(teamRepository, 'getTeamUser').mockResolvedValue(teamUser);
+
+        await service.changeTeamManager(team.id, user.uuid);
+
+        expect(teamRepository.updateById).toHaveBeenCalledWith(team.id, {
+          managerId: user.uuid,
+        });
+      });
+    });
+
+    describe('deleteTeam', () => {
+      it('When team is not found, then it should throw', async () => {
+        const teamId = 'team-id';
+
+        jest.spyOn(teamRepository, 'getTeamById').mockResolvedValue(null);
+
+        await expect(service.deleteTeam(teamId)).rejects.toThrow(
+          BadRequestException,
+        );
+      });
+
+      it('When team is default team, then it should throw', async () => {
```

Comment:
_When the team to update is the default one of a workspace_ could be a better legend